File: samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbPowerPacksShapeSuspendPaint/VB/ShapeSuspendPaint.vb
Public Class ShapeSuspendPaint
    ' <Snippet1>
    Private Sub OvalShape1_Click() Handles OvalShape1.Click
        ' Suspend painting.
        OvalShape1.SuspendPaint()
        ' Set some properties.
        OvalShape1.BackStyle = PowerPacks.BackStyle.Opaque
        OvalShape1.BackColor = Color.Blue
        OvalShape1.FillStyle = PowerPacks.FillStyle.Plaid
        OvalShape1.FillColor = Color.Red
        ' Resume painting and execute any pending requests.
        OvalShape1.ResumePaint(True)
    End Sub
    ' </Snippet1>
End Class

File: samples/snippets/csharp/VS_Snippets_VBCSharp/VbPowerPacksShapeSuspendPaint/CS/ShapeSuspendPaint.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Microsoft.VisualBasic.PowerPacks;

namespace VbPowerPacksShapeSuspendPaintCS
{
    public partial class ShapeSuspendPaint : Form
    {
        public ShapeSuspendPaint()
        {
            InitializeComponent();
        }

        // <Snippet1>
        private void ovalShape1_Click(System.Object sender, System.EventArgs e)
        {
            // Suspend painting.
            ovalShape1.SuspendPaint();
            // Set some properties.
            ovalShape1.BackStyle = BackStyle.Opaque;
            ovalShape1.BackColor = Color.Blue;
            ovalShape1.FillStyle = FillStyle.Plaid;
            ovalShape1.FillColor = Color.Red;
            // Resume painting and execute any pending requests.
            ovalShape1.ResumePaint(true);
        }
        // </Snippet1>
    }
}
